### PR TITLE
feat: warn if --quantile_tau is not used

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

File: vowpalwabbit/parse_args.cc [hunk 1]
```diff
@@ -1048,6 +1048,12 @@ void parse_example_tweaks(options_i& options, VW::workspace& all)
   }
 
   all.loss = getLossFunction(all, loss_function, loss_parameter);
+  if (options.was_supplied("quantile_tau") && all.loss->getType() != "quantile")
```
Comment (lalo) [inline vowpalwabbit/parse_args.cc:1051]: why would this happen though?

Comment (lalo) [reply vowpalwabbit/parse_args.cc:1051]: and why only test for quantile_tau, I'm assuming this can happen with any other loss related arg that is later overridden somehow

Comment (jackgerrits) [reply vowpalwabbit/parse_args.cc:1051]: quantile is the only loss function which accepts an argument.

**Review by lalo — APPROVED** [review]
